### PR TITLE
Generated Protocol Bindings

### PR DESCRIPTION
**Pull request by RXminuS**
Adds generated protocol bindings for use in #1921  

## Test plan

- Doesn't use generated protocol bindings. Verified that build still works

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -341,6 +345,57 @@ tasks {
     return buildCodyDir
   }
 
+  fun copyProtocol(): Unit {
+    // This is just a temporary solution to make it easier to use certain generated protocol
+    // messages.
+    // Ultimately this should entirely be replaced by use of the generated protocol.
+    if (!isForceProtocolCopy) {
```
Comment (pkukielka) [inline build.gradle.kts:352]: What is the purpose of this flag if it needs to always be set for task to have any effect?
Shouldn't force be necessary only if the protocol files already exists? (is it even an option that no such files would be present)?
If that flag is always needed then equally good we can just not have it.

**Review by pkukielka — APPROVED** [review]
LGTM except one other comment.

**Review by mkondratek — APPROVED** [review]
ok but pls note the Peter's comment

File: build.gradle.kts (hunk 1 above)
Comment (RXminuS) [reply build.gradle.kts:352]: I'll address this when we set up CI & automatic generation on builds etc. For now it's all manual while we iron out the wrinkles

**Pull request merged**